frontend/src/scenes/billing/Plan.tsx

```diff
@@ -0,0 +1,104 @@
+import { Skeleton } from 'antd'
+import React, { useEffect, useState } from 'react'
+import { PlanInterface } from '~/types'
+import clsx from 'clsx'
+import { LemonButton, LemonDivider } from '@posthog/lemon-ui'
+import { IconLock } from 'lib/components/icons'
+
+export function Plan({
+    plan,
+    onSubscribe,
+    currentPlan,
+    canHideDetails = true,
+    primaryCallToAction = true,
+}: {
+    plan: PlanInterface
+    canHideDetails?: boolean
+    onSubscribe?: (plan: PlanInterface) => void
+    currentPlan?: boolean
+    primaryCallToAction?: boolean
+}): JSX.Element {
+    const [detail, setDetail] = useState('')
+    const [isDetailLoading, setIsDetailLoading] = useState(true)
+    const [showDetails, setShowDetails] = useState(false)
+
+    const loadPlanDetail = async (key: string): Promise<void> => {
+        const response = await fetch(`/api/plans/${key}/template/`)
+        if (response.ok) {
+            setDetail(await response.text())
+        }
+        setIsDetailLoading(false)
+    }
+
+    useEffect(() => {
+        loadPlanDetail(plan.key)
+    }, [plan.key])
```

Comment:
Not necessarily for this PR, but I suspect this would be cleaner as a `planDetailHtml` loader in `billingLogic` instead of a hook